probe/resolver.go

```diff
@@ -92,13 +92,15 @@ func (r staticResolver) resolveHosts() {
 			}
 		}
 
+		endpoints := make([]string, 0, len(addrs))
 		for _, addr := range addrs {
 			// For now, ignore IPv6
 			if addr.To4() == nil {
 				continue
 			}
-			r.add(net.JoinHostPort(addr.String(), peer.port))
+			endpoints = append(endpoints, net.JoinHostPort(addr.String(), peer.port))
 		}
+		r.set(peer.hostname, endpoints)
```

Comment:
Hmm a temporary glitch in DNS will stop us pushing to a app for ~1 min.  I guess you're doing this to stop us trying to push to apps which have gone away?  I think we probably want to decouple some sort of app-timeout from the resolving (perhaps this should instead be coupled to the backoff?  n strikes at backoff then it gets removed?)